internal/postgres/statement.go

```diff
@@ -0,0 +1,697 @@
+	if n.Relation == nil {
+		statementError(conv, n, fmt.Errorf("relation is nil"))
+		return nil
+	}
+	pgTable, spTable, err := getTableName(conv, *n.Relation)
+	if err != nil {
+		statementError(conv, n, fmt.Errorf("can't get table name: %w", err))
+		return nil
+	}
+	conv.statsAddRow(spTable, conv.schemaMode())
+	cols, err := getCols(conv, pgTable, n.Cols.Items)
+	if err != nil {
+		statementError(conv, n, fmt.Errorf("can't get col name: %w", err))
+		conv.statsAddBadRow(spTable, conv.schemaMode())
+		return nil
+	}
+	var values []string
+	switch sel := n.SelectStmt.(type) {
+	case nodes.SelectStmt:
+		values = getVals(conv, sel.ValuesLists, n)
+		conv.dataStatement([]nodes.Node{n})
+		if conv.dataMode() {
+			return &copyOrInsert{stmt: insert, spTable: spTable, pgTable: pgTable, cols: cols, vals: values}
+		}
+	default:
+		conv.unexpected(fmt.Sprintf("Found %s node while processing InsertStmt SelectStmt", prNodeType(sel)))
+	}
+	return nil
+}
+
+func processCopyStmt(conv *Conv, n nodes.CopyStmt) *copyOrInsert {
+	// Always return a copyOrInsert{stmt: copyFrom, ...} even if we
+	// encounter errors. Otherwise we won't be able to parse
+	// the data portion of the COPY-FROM statement, and we'll
+	// likely get stuck at this point in the pg_dump file.
+	pgTable := "BOGUS_COPY_FROM_TABLE"
+	spTable := "BOGUS_COPY_FROM_TABLE"
+	var err error
+	if n.Relation != nil {
+		pgTable, spTable, err = getTableName(conv, *n.Relation)
+		if err != nil {
+			conv.unexpected(fmt.Sprintf("Processing %v statement: %s", reflect.TypeOf(n), err))
+		}
+	} else {
+		statementError(conv, n, fmt.Errorf("relation is nil"))
+	}
+	var cols []string
+	for _, a := range n.Attlist.Items {
+		s, err := getString(a)
+		if err == nil {
+			s, err = GetSpannerCol(conv, pgTable, s, true)
+		}
+		if err != nil {
+			conv.unexpected(fmt.Sprintf("Processing %v statement Attlist: %s", reflect.TypeOf(n), err))
+			s = "BOGUS_COPY_FROM_COLUMN"
+		}
+		cols = append(cols, s)
+	}
+	conv.dataStatement([]nodes.Node{n})
+	return &copyOrInsert{stmt: copyFrom, spTable: spTable, pgTable: pgTable, cols: cols}
+}
+
+func processVariableSetStmt(conv *Conv, n nodes.VariableSetStmt) {
+	if n.Name != nil && *n.Name == "timezone" {
+		if len(n.Args.Items) == 1 {
+			switch c := n.Args.Items[0].(type) {
+			case nodes.A_Const:
+				tz, err := getString(c.Val)
+				if err != nil {
+					statementError(conv, n, fmt.Errorf("can't get Arg: %w", err))
+					return
+				}
+				loc, err := time.LoadLocation(tz)
+				if err != nil {
+					statementError(conv, n, err)
+					return
+				}
+				conv.SetLocation(loc)
+			default:
+				statementError(conv, n, fmt.Errorf("found %s node in Arg", reflect.TypeOf(c)))
+				return
+			}
+		}
+	}
+}
+
+// getSpannerType determines the Spanner type for the scalar postgres type
+// defined by id and mods. This is the core postgres-to-Spanner type mapping.
+// getSpannerType returns the Spanner type and a bool to indicate whether
```

Comment:
`// getSpannerType returns the Spanner type and a bool`
No it doesn't?

Reply:
Fixed.
Thanks for catching!